vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/condition/VistaProblemToR4ConditionTransformer.java

```
@@ -117,39 +120,61 @@ CodeableConcept code(Problems.Problem rpcCondition) {
-                .source(site)
-                .lastUpdated(toHumanDateTime(rpcCondition.updated()))
-                .build())
-        .clinicalStatus(clinicalStatus(rpcCondition))
-        .category(category())
-        .code(code(rpcCondition))
-        .subject(toReference("Patient", patientIcn, null))
-        .verificationStatus(verificationStatus(rpcCondition.unverified()))
-        .onsetDateTime(toHumanDateTime(rpcCondition.onset()))
-        .recordedDate(toHumanDateTime(rpcCondition.entered()))
-        .build();
+    return Annotation.builder()
+        .text(comment.commentText())
+        .time(toHumanDateTimeFromString(comment.entered()))
+        .authorString(comment.enteredBy())
+        .build()
+        .asList();
   }
 
   Stream<Condition> toFhir() {
-    return rpcResults.problemStream().map(this::toCondition).filter(Objects::nonNull);
+    if (isBlank(vistaProblem)) {
+      return Stream.empty();
+    }
+    return Stream.of(
+        Condition.builder()
+            .id(idFrom(vistaProblem.id()))
+            .meta(
+                Meta.builder()
+                    .source(site)
+                    .lastUpdated(toHumanDateTimeFromValueOnlyXmlAttribute(vistaProblem.updated()))
+                    .build())
+            .clinicalStatus(clinicalStatus(vistaProblem))
+            .category(category())
+            .code(code(vistaProblem))
+            .note(emptyToNull(note(vistaProblem.comment())))
+            .subject(toReference("Patient", patientIcn, null))
+            .verificationStatus(verificationStatus(vistaProblem.unverified()))
+            .onsetDateTime(valueOnlyXmlAttributeDateTime(vistaProblem.onset()))
+            .recordedDate(valueOnlyXmlAttributeDateTime(vistaProblem.entered()))
+            .build());
+  }
+
+  private String valueOnlyXmlAttributeDateTime(ValueOnlyXmlAttribute onset) {
+    try {
+      return toHumanDateTimeFromValueOnlyXmlAttribute(onset);
+    } catch (DateTimeException e) {
+      return null;
```

Should we just squash all exceptions here?

---

We know what exception will be thrown and Fortify isn't a fan of broadly catching exceptions.